include/podio/ROOTNTupleWriter.h

```diff
@@ -0,0 +1,57 @@
+#ifndef PODIO_ROOTNTUPLEWRITER_H
+#define PODIO_ROOTNTUPLEWRITER_H
+
+#include "podio/CollectionBase.h"
+#include "podio/GenericParameters.h"
+#include "podio/utilities/DatamodelRegistryIOHelpers.h"
+#include "podio/Frame.h"
+
+#include "TFile.h"
+#include <ROOT/RNTuple.hxx>
+#include <ROOT/RNTupleModel.hxx>
+
+#include <string>
+#include <vector>
+#include <iostream>
+#include <string_view>
+
+namespace podio {
+
+class ROOTNTupleWriter {
+public:
+  ROOTNTupleWriter(const std::string& filename);
+  ~ROOTNTupleWriter();
+
+  ROOTNTupleWriter(const ROOTNTupleWriter&) = delete;
+  ROOTNTupleWriter& operator=(const ROOTNTupleWriter&) = delete;
+
+  void writeFrame(const podio::Frame& frame, const std::string& category);
+  void writeFrame(const podio::Frame& frame, const std::string& category, const std::vector<std::string>& collsToWrite);
+  void finish();
+
+private:
+
+  using StoreCollection = std::pair<const std::string&, podio::CollectionBase*>;
+  std::unique_ptr<ROOT::Experimental::RNTupleModel> createModels(const std::vector<StoreCollection>& collections);
+
+  std::unique_ptr<ROOT::Experimental::RNTupleModel> m_metadata{};
+  std::map<std::string, std::unique_ptr<ROOT::Experimental::RNTupleWriter>> m_writers{};
+  std::unique_ptr<ROOT::Experimental::RNTupleWriter> m_metadataWriter{};
+
+  std::unique_ptr<TFile> m_file{};
+
+  DatamodelDefinitionCollector m_datamodelCollector{};
+
+  std::map<std::string, std::vector<int>> m_collectionId{};
+  std::map<std::string, std::vector<std::string>> m_collectionName{};
+  std::map<std::string, std::vector<std::string>> m_collectionType{};
+  std::map<std::string, std::vector<bool>> m_isSubsetCollection{};
+  std::set<std::string> m_categories{};
```

Comment:
I suppose many of these maps (and a few of the above ones) are here to support writing multiple categories? Wouldn't it be simpler to have a small(-ish) helper struct that bundles all the information for one category and one map for name -> struct? Similar to what is done in the ROOTFrameWriter/Reader?

Reply:
Now they are grouped, with the possibility of grouping a few more things that are used in the reader too